components/omega/src/ocn/Eos.h

```diff
@@ -53,22 +53,20 @@ class TEOS10Poly75t {
       }
    }
 
-   // To be added:
    // Note that it assumes that we have called calcPCoeffs already
    //
-   // KOKKOS_FUNCTION void calcDisplacedSpecVol(const Array2DReal
-   // &SpecVolDisplaced,
-   //                                 I4 ICell, I4 KChunk,
-   //                                 const Array2DReal &Pressure) const {
-
-   //    const I4 KStart = KChunk * VecLength;
-   //    // insert exception for the surface
-   //    for (int KVec = 0; KVec < VecLength; ++KVec) {
-   //       const I4 K        = KStart + 1 + KVec;
-   //       SpecVolDisplaced(ICell, K) = calcRefProfile(Pressure(ICell, K-1)) +
-   //       calcDelta(Pressure(ICell, K-1));
-   //    }
-   // }
+   KOKKOS_FUNCTION void
+   calcDisplacedSpecVol(const Array2DReal &SpecVolDisplaced, I4 ICell,
+                        I4 KChunk, const Array2DReal &Pressure) const {
+
+      const I4 KStart = KChunk * VecLength;
+      for (int KVec = 0; KVec < VecLength; ++KVec) {
+         const I4 K                 = KStart + KVec;
+         const I4 KTmp              = Kokkos::min(K + 1, 60);
```

Comment:
Let's not hardcode 60 here and use nVertLevels instead. We could make nVertLevels a member variable and set it in the constructor.

Reply:
ok, it took me a while to figure out the syntax for the constructor, but got that sorted.